src/middleware/auth.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+'use strict';
+
+const jwt = require('jwt-simple');
+const wrap = require('co-express');
+const createError = require('http-errors');
+const _ = require('lodash');
+
+const EXPIRED = 'Token expired';
+const VALIDATION_FAILED = 'Token validation failed';
+const NOT_CONNECTED = 'User not connected';
+const NEED_KEY = 'IDP secret for JWT is not set';
+const ALGORITHM = 'RS256';
+const KEY_END_STRING = '-----END PUBLIC KEY-----\n';
+
+/**
+ * This functions takes a string of concatenated PEM public keys and splits
+ * them in an array.
+ * @param {string} keysString - Concatenated PEM public keys
+ * @returns {Array} Separated PEM public keys
+ */
+function retrieveKeys(keysString) {
+  const keys = keysString.split(KEY_END_STRING);
+
+  return _.map(keys, (item, index) => (index + 1 === keys.length) ? item : item + '-----END PUBLIC KEY-----');
+}
+
+/**
+ * This functions checks several PEM public keys among the
+ * ones defined in the keys array. This functions executes jwt.decode on
+ * every available PEM public keys and then returns the payload if there is a matching key.
+ * Otherwise it throws and error. Among these errors, the 'Token expired' one
+ * is threw in priority because it has more details than the 'Token validation failed' error.
+ * @param {array} keys - PEM public keys
+ * @param {string} token - Issuer JWT token
+ * @returns {object} The token's payload
+ * @throws Custom error with an HTTP code and message
+ */
+function validateToken(keys, token) {
+  let payload;
+  let error;
+
+  _.map(keys, key => {
+    try {
+      payload = jwt.decode(token, key, false, [ALGORITHM]);
+    } catch (err) {
+      if (err.message === EXPIRED) {
+        error = createError(401, EXPIRED);
```

Comment:
`http-errors` give you some syntax sugar to avoid using http status code directly `createError.Unauthorized(EXPIRED )`